### PR TITLE
feat: custom trade market sell amount buttons

### PR DESCRIPTION
**Pull request by byBackfish**
Implements [add-more-options-in-the-selling-menu-of-tm](https://feedback.wynntils.com/posts/103/add-more-options-in-the-selling-menu-of-tm)

Adds 3 buttons to the trade market gui, pressing them will automatically put the set amount (in the config) into the chat. 
Fixes various bugs
TradeMarketOverlay now uses the nbt tag to determine the amount of the item to sell

### CODE REVIEW

**Comment by byBackfish** [comment]
image

File: src/main/java/com/wynntils/modules/utilities/configs/UtilitiesConfig.java [hunk 1]
```diff
@@ -495,6 +495,16 @@ public static class Market extends SettingsClass {
...
         public boolean showSellAllButton = true;
+
+        @Setting(displayName = "Custom Sell button 1 amount", description = "What amount should be selected when the custom sell button 1 is clicked?\n\n§8Set the value to 0 to disable this")
+        public String customSellButton1 = "64";
+
+        @Setting(displayName = "Custom Sell button 2 amount", description = "What amount should be selected when the custom sell button 2 is clicked?\n\n§8Set the value to 0 to disable this")
+        public String customSellButton2 = "1728";
+
+        @Setting(displayName = "Custom Sell button 3 amount", description = "What amount should be selected when the custom sell button 3 is clicked?\n\n§8Set the value to 0 to disable this")
+        public String customSellButton3 = "6399";
```
Comment (kristofbolyai) [inline src/main/java/com/wynntils/modules/utilities/configs/UtilitiesConfig.java:506]: String for an integer setting?

Comment (kristofbolyai) [reply src/main/java/com/wynntils/modules/utilities/configs/UtilitiesConfig.java:506]: I assume the problem is that integer configs don't really render properly without a limit, due to how the config GUI is written?

**Review by kristofbolyai — COMMENTED** [review]
Also, was the bug fixed where the sell all button inputted numbers even when the user did not press the button?

File: src/main/java/com/wynntils/modules/utilities/configs/UtilitiesConfig.java (hunk 1 above)
Comment (byBackfish) [reply src/main/java/com/wynntils/modules/utilities/configs/UtilitiesConfig.java:506]: yea, and if i set a limit (with the max bound of 6399, which is the trade market sell limit), it goes up in steps of 30+

**Comment by byBackfish** [comment]
yea, that bug was fixed already

File: src/main/java/com/wynntils/modules/utilities/configs/UtilitiesConfig.java (hunk 1 above)
Comment (kristofbolyai) [reply src/main/java/com/wynntils/modules/utilities/configs/UtilitiesConfig.java:506]: aah. this is sad

Comment (kristofbolyai) [reply src/main/java/com/wynntils/modules/utilities/configs/UtilitiesConfig.java:506]: We can limit the string field to numbers only though by checking config updates and reverting if a non number char was added.

Comment (byBackfish) [reply src/main/java/com/wynntils/modules/utilities/configs/UtilitiesConfig.java:506]: i assume with `onSettingChanged`? How would i revert/cancel it tho? Does it save the previous value anywhere?

Comment (kristofbolyai) [reply src/main/java/com/wynntils/modules/utilities/configs/UtilitiesConfig.java:506]: I think it does in arguments? If it does not, let's leave it like this.

Comment (byBackfish) [reply src/main/java/com/wynntils/modules/utilities/configs/UtilitiesConfig.java:506]: the only argument is `name`, so sadly not

File: src/main/java/com/wynntils/modules/utilities/overlays/inventories/TradeMarketOverlay.java [hunk 2]
```diff
@@ -117,7 +138,24 @@ private String getItemName(InventoryBasic inventory) {
         Matcher matcher = ITEM_NAME_PATTERN.matcher(itemStack.getDisplayName());
 
         if (!matcher.matches()) return null;
-        return matcher.group(1);
+        return matcher.group(2);
+    }
+
+    private void addCustomSellItems(String itemName, InventoryBasic inventory) {
+        if (StringUtils.parseIntOr(UtilitiesConfig.Market.INSTANCE.customSellButton1, 0) > 0) {
```
Comment (kristofbolyai) [inline src/main/java/com/wynntils/modules/utilities/overlays/inventories/TradeMarketOverlay.java]: Don't parse everything twice?

**Review by kristofbolyai — DISMISSED** [review]

File: src/main/java/com/wynntils/modules/utilities/configs/UtilitiesConfig.java (hunk 1 above)
Comment (HighCrit) [inline src/main/java/com/wynntils/modules/utilities/configs/UtilitiesConfig.java:506]: Is there demand for this?

Comment (byBackfish) [reply src/main/java/com/wynntils/modules/utilities/configs/UtilitiesConfig.java:506]: Yes, it's very convenient, especially for people that grind professions. They often sell the max amount possible on the trade market (6399) and this makes it a bit easier

**Comment by ryanzhoudev** [comment]
I think the button and text for this is rather large, is the settings text necessary?
image

**Comment by ryanzhoudev** [comment]
> I think the button and text for this is rather large, is the settings text necessary?

Also, I think the "Click to sell all items in your inventory" shouldn't be there

**Review by ryanzhoudev — DISMISSED** [review]

**Review by magicus — DISMISSED** [review]

**Review by magicus — APPROVED** [review]

**Review by kristofbolyai — APPROVED** [review]

**Pull request merged**